src/components/DataTable/DataTable.tsx

```diff
@@ -159,129 +176,141 @@ export function DataTable<T>({
-                                cell.getContext(),
-                              )}
-                            </td>
-                          ))}
-                        </DataTableRow>
-                      ))}
-                    </>
-                  ) : (
-                    <DataTableRow
-                      isInteractive={isInteractive}
-                      isSelected={row.getIsSelected()}
-                    >
-                      {row.getVisibleCells().map((cell) => (
-                        <td
-                          className={styles['data-table__cell-container']}
-                          key={cell.id}
-                        >
-                          {flexRender(
-                            cell.column.columnDef.cell,
-                            cell.getContext(),
-                          )}
-                        </td>
-                      ))}
-                    </DataTableRow>
-                  )}
-                </React.Fragment>
-              );
-            })}
-          </tbody>
-        </DataTableTable>
-      )}
-    </div>
+                    {caption}
+                  </div>
+                )}
+                {subcaption && (
+                  // TODO: Warn when only using subcaption
+                  <div
+                    aria-hidden="true"
+                    className={styles['data-table__subcaption']}
+                  >
+                    {subcaption}
+                  </div>
+                )}
+              </div>
+            )}
+            {onSearchChange && (
+              <div className={styles['data-table__search']}>
+                <DataTableSearch />
+              </div>
+            )}
+            {actions && (
+              <div className={styles['data-table__actions']}>
+                <DataTableActions>{actions}</DataTableActions>
+              </div>
+            )}
+          </div>
+        )}
+        {/* Provide an escape hatch for specifying all aspects of a table using `children` and Sub-components directly */}
+        {children ?? (
+          <DataTableTable
+            className={tableClassName}
+            rowStyle={rowStyle}
+            size={size}
+            tableStyle={tableStyle}
+          >
+            {(caption || subcaption) && (
+              <caption className={styles['data-table__aria-caption']}>
+                {`${caption}${subcaption ? ': ' + subcaption : ''}`}
+              </caption>
+            )}
+            <DataTableHeader>
+              {table?.getHeaderGroups().map((headerGroup) => (
+                <DataTableRow key={headerGroup.id}>
+                  {headerGroup.headers.map((header) => {
+                    // TODO-AH: figure out better way to detect a non-default column size
+                    const columnWidth =
+                      header.getSize() !== 150
+                        ? `${header.getSize()}px`
+                        : undefined;
```

Comment:
Need to investigate how TanStack Table sets the column width. `.getSize` is always defined, and defaults to 150px. Marking this value as special here b/c all sizings will be based on token unit sizes (16 * x) and 150 doesn't land on one of those boundaries (... 144, 160, 176, ...)